### PR TITLE
4/ Implemented entity conversion

### LINKED ISSUE #191: Convert EnemyNPC to FriendlyNPC once the enemy is beaten
# **Description**
**Task:** Once an EnemyNPC is beaten, change the entity type of the beaten NPC to a FriendlyNPC
**Feature:** Friendly Animals (NPC) [Team 4]

This task involves modifying the game's logic to recognise when an EnemyNPC has been defeated. Upon defeat, the EnemyNPC's entity type will be updated to FriendlyNPC. This change will trigger corresponding behaviors and interactions, such as preventing the FriendlyNPC from attacking the player and potentially introducing new gameplay elements or interactions. This feature will add a dynamic element to the game world by allowing defeated enemies to become potential allies or sources of information.

# **Dependencies**
- Team 5 for Combat Interaction
 
# **Milestones**
List of steps that need to be completed for this task.
- [x] Implement a class for the conversion of the entities (Sep. 2)
- [x] Implement a check to see if an NPC has been defeated (Sep. 5)
- [x] Convert EnemyNPC to FriendlyNPC (Sep. 5)
- [x] Implement testing for the conversion of entities (Sep. 9)
- [x] Write up the documentation for the new class in the wiki (Sep. 10)
 
**Completion Deadline:** Sep. 11
 
# **Documentation**
[Entity Conversion Details](https://github.com/UQcsse3200/2024-studio-2/wiki/Enemy-Converting-to-Friendly-Upon-Defeat)
 
# **Member**
- Shubh Gupta (@shubhgupta2510)

### PR DESCRIPTION
**Pull request by shubhgupta2510**
Closes #191 and #277 and implements the conversion of enemy entities to friendly entities once defeated. Currently, once the enemy is defeated, the game goes to a "Game Over" screen but the EntityConverter class works as expected and should show the enemy as a friendly once defeated.

### CODE REVIEW

**Review by Jul-Wong — COMMENTED** [review]
I think the item listener needs to be uncommented, and some commented out methods should probably be deleted if not used.

File: source/core/src/main/com/csse3200/game/areas/ForestGameArea.java [hunk 1]
```diff
@@ -48,7 +48,7 @@ public class ForestGameArea extends GameArea {
   private static final GridPoint2 KANGAROO_BOSS_SPAWN = new GridPoint2(25, 10);
   private static final float WALL_WIDTH = 0.1f;
   private final TerrainFactory terrainFactory;
-  private final List<Entity> enemies;
+  public static List<Entity> enemies = null;
```
Comment (owenrharding) [inline source/core/src/main/com/csse3200/game/areas/ForestGameArea.java:51]: I wouldn't set it to null if you're going to set it to a new ArrayList in the initialiser anyways

**Review by owenrharding — COMMENTED** [review]
Looks good, just merge with main and I'll approve

**Review by LucasRyall — CHANGES_REQUESTED** [review]
Fatal Bug that needs to be resolved, on returning to the MainGameScreen enemies aren't properly disposed of, and cause the game to crash (unknown exact cause but replicatable by killing an enemy and then walking over their last pos/spawnpos).

**Pull request closed without merging**